api/src/main/java/io/opentelemetry/trace/Span.java

```diff
@@ -205,6 +205,18 @@
    */
   boolean isRecordingEvents();
 
+  /**
+   * Returns an {@code Object} used for backwards interoperability with either OpenTracing or
+   * OpenCensus. Do not use this object. Scheduled to be removed in the next version.
+   */
+  Object getImplObj();
```

Comment:
should we mark these methods as `@deprecated`, since they're going away?

Reply:
Yes, **if** we actually happen to add them (which I bet is going to be polemical, hah ;) )

Reply:
I would be surprised if it would be polemic.. more likely everyone just doesn't like it.

Reply:
Polemic because I'd be pushing for it and people would be disliking it ;)

Reply:
(because this is literally adding special-case hacks into the api package for one optional feature of OT.. I doubt anyone will actually like this. polemic being usually used when there are those who love or hate something.. I'd doubt there are lovers, but have been surprised before)

Reply:
Well, it's not an optional feature of OT, as this is something part of the OT standard. There have been discussion about not supporting a few cases, but that's something we still need to decide (I'm wondering how many people's code we would be breaking).

I don't like it neither, but it's something that could help us **a lot** in the transition.

Reply:
OT is not a standard

Reply:
also by optional, the list of things supported don't all support this. there's no compatability chart either. for example https://github.com/apache/skywalking/blob/master/apm-application-toolkit/apm-toolkit-opentracing/src/main/java/org/apache/skywalking/apm/toolkit/opentracing/ByteBufferContext.java

Reply:
so basically, I think you need to be honest with yourself that OT has no means of knowing or even tracking which features are required vs not. Knowing that, adding a hack for this feature is dubious especially as the feature itself is dodgy. unsolicited 2p is find another way

Reply:
We want to help users and want to provide as many features as possible, even if we don't know for sure what features are more used than others. That being said, `SpanContext` containing Baggage has been present for **long time** in OT, and I know Jaeger supports that, for a start.

And it was been a tricky feature, and that's why we are not following the same path in OTel (by decoupling `SpanContext` and the notion of Baggage/TagMap), but we still want to help our users in case they have need this one. But because of the implementation nature, I'm precisely posting this as a prototype ;)

Reply:
main point of all of this: escalate to some other library designers in the OT inner circle as I'm sure there are alternative approaches. I know you want the feature to work, and are less concerned on the how. Another way surely can exist. There are several expert vendors associated with OT, it would be surprising if not one could offer another idea.